{"name": "test_get[djkfhdsjfdss]", "status": "broken", "statusDetails": {"message": "TypeError: unsupported operand type(s) for +: 'int' and 'str'", "trace": "self = <api.department.Department object at 0x00000218A66A1B70>\nsteps = [{'method': 'tag.get', 'userid': '${userid}'}, {'extract': 'before', 'path': '$..userid'}, {'assertion': 'dfmfdsdfdddfsffddf'}, {'assertion': 0, 'path': '$..errcode'}, {'assertion': 'ok', 'path': '$..errmsg'}]\n\n    def api_steps(self, steps: list):\n        for step in steps:\n            raw = yaml.dump(step)\n            for key, value in self.params.items():\n                raw = raw.replace(f'${{{key}}}', repr(value))\n            step = yaml.safe_load(raw)\n            if isinstance(step, dict):\n                if 'method' in step.keys():\n                    Get_log().log_info('正在测试'+ step['method'])\n                    allure.attach(step['method']+'测试')\n                    method = step['method'].split('.')[-1]\n                    getattr(self, method)(**step)\n                if \"path\" in step.keys():\n                    self.code = getattr(self, 'jsonpath')(**step)\n                if 'assertion' in step.keys():\n                    try:\n>                       assert self.code == step['assertion']\nE                       AssertionError\n\n..\\api\\baseapi.py:68: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <testcase.test_department.TestDepartment object at 0x00000218A66A1F98>\nuserid = 'djkfhdsjfdss'\n\n    @allure.step('测试')\n    @pytest.mark.parametrize('userid',['dfmfdsdfdddfsffddf','djkfhdsjfdss'])\n    def test_get(self,userid):\n        self.department.params['userid'] =userid\n>       self.department.api_steps(self.data)\n\ntest_department.py:16: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <api.department.Department object at 0x00000218A66A1B70>\nsteps = [{'method': 'tag.get', 'userid': '${userid}'}, {'extract': 'before', 'path': '$..userid'}, {'assertion': 'dfmfdsdfdddfsffddf'}, {'assertion': 0, 'path': '$..errcode'}, {'assertion': 'ok', 'path': '$..errmsg'}]\n\n    def api_steps(self, steps: list):\n        for step in steps:\n            raw = yaml.dump(step)\n            for key, value in self.params.items():\n                raw = raw.replace(f'${{{key}}}', repr(value))\n            step = yaml.safe_load(raw)\n            if isinstance(step, dict):\n                if 'method' in step.keys():\n                    Get_log().log_info('正在测试'+ step['method'])\n                    allure.attach(step['method']+'测试')\n                    method = step['method'].split('.')[-1]\n                    getattr(self, method)(**step)\n                if \"path\" in step.keys():\n                    self.code = getattr(self, 'jsonpath')(**step)\n                if 'assertion' in step.keys():\n                    try:\n                        assert self.code == step['assertion']\n                    except Exception as e:\n>                       Get_log().log_error(self.code + '！=' + step['assertion'])\nE                       TypeError: unsupported operand type(s) for +: 'int' and 'str'\n\n..\\api\\baseapi.py:70: TypeError"}, "steps": [{"name": "测试", "status": "broken", "statusDetails": {"message": "TypeError: unsupported operand type(s) for +: 'int' and 'str'\n", "trace": "  File \"D:\\python\\lib\\site-packages\\allure_commons\\_allure.py\", line 172, in impl\n    return func(*a, **kw)\n  File \"H:\\练习\\test_reques_wework\\testcase\\test_department.py\", line 16, in test_get\n    self.department.api_steps(self.data)\n  File \"H:\\练习\\test_reques_wework\\api\\baseapi.py\", line 70, in api_steps\n    Get_log().log_error(self.code + '！=' + step['assertion'])\n"}, "attachments": [{"source": "b9e70150-8609-4302-8145-b2f0038a4b37-attachment.attach"}], "parameters": [{"name": "userid", "value": "'djkfhdsjfdss'"}], "start": 1634408272948, "stop": 1634408274403}], "attachments": [{"name": "log", "source": "d6e5c712-2065-40fe-9aec-989162f3ba41-attachment.txt", "type": "text/plain"}], "parameters": [{"name": "userid", "value": "'djkfhdsjfdss'"}], "start": 1634408272948, "stop": 1634408274403, "uuid": "9722bc9c-79b7-4bd3-be8a-84ced0a6cf61", "historyId": "5c3bc40267495b9ef4867fee7c6d02db", "testCaseId": "dcccf3236d29225883a842a7b88face6", "fullName": "testcase.test_department.TestDepartment#test_get", "labels": [{"name": "parentSuite", "value": "testcase"}, {"name": "suite", "value": "test_department"}, {"name": "subSuite", "value": "TestDepartment"}, {"name": "host", "value": "PC-202008151337"}, {"name": "thread", "value": "3252-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "testcase.test_department"}]}